### PR TITLE
Chain service: Add support for an optional fallback mempool.space URL

### PR DESCRIPTION
**Pull request by ok300**
This PR adds an optional fallback URL for a 2nd mempool.space instance.

It is called automatically if the call to the main mempool instance fails. If the fallback call also fails, the error is propagated to the caller.

### CODE REVIEW

**Comment by dangeross** [comment]
Should we seek permission to use Emzy`s mempool.space instance as the fallback? Maybe a better alternative is to set the default to None and allow devs to set the fallback as they see fit

**Comment by dangeross** [comment]
Code LGTM, I just question if the `mempoolspace_fallback_url` should be optional in the Config and that it should be set by the developer using the SDK. If mempool.space is down, there is then network load going another single point of failure.

**Comment by ok300** [comment]
The fallback is more useful if the default has a value. And Emzy's instance is public, so it seems reasonable we could start using it. But I'm flexible on both points.

As background: in my tests, I've seen short periods of HTTP 404 or 503 errors every now and then on the main mempool instance. It usually lasts only a few seconds, then it works again. This PR is an enhancement proposal for to better handle these kinds of cases.

File: libs/sdk-core/src/chain.rs [hunk 1]
```diff
@@ -124,6 +125,7 @@ pub(crate) fn get_utxos(address: String, transactions: Vec<OnchainTx>) -> Result
 #[derive(Clone)]
 pub(crate) struct MempoolSpace {
     pub(crate) base_url: String,
+    pub(crate) fallback_base_url: Option<String>,
```
Comment (dangeross) [inline libs/sdk-core/src/chain.rs:128]: Is there any point `fallback_base_url` being optional if the Config `mempoolspace_fallback_url` is not optional?

**Comment by yaslama** [comment]
If we want to able to control from Breez the default url, the best way way is to add a rpc function (into beez server) which returns a list of default urls. If no url is configured, the sdk calls this rpc and tries the first one and if it fails the second one, etc..

File: libs/sdk-core/src/chain.rs (hunk 1 above)
Comment (ok300) [reply libs/sdk-core/src/chain.rs:128]: Good point, fixed.

**Comment by roeierez** [comment]
@ok300 if we decided to take the values from the server, can we close this one?

**Comment by ok300** [comment]
Yes. I added an issue in the LSPD repo instead.

**Comment by yaslama** [comment]
> Yes. I added an issue in the LSPD repo instead.

Why in lspd? It's in breez server
https://github.com/breez/breez-sdk/pull/708#issuecomment-1880068799

**Comment by ok300** [comment]
I thought the idea is who runs an LSP may want to control the default mempool instances for their users.

I moved it to the breez server repo: https://github.com/breez/server/issues/41

**Pull request closed without merging**